src/main/kotlin/org/rust/ide/template/postfix/stringBasedPostfixTemplates.kt

```diff
@@ -61,3 +67,29 @@ class MatchPostfixTemplate(provider: RsPostfixTemplateProvider) :
         }
     }
 }
+
+class IterPostfixTemplate(provider: RsPostfixTemplateProvider) :
```

Comment:
IMO it's better to split it into two classes like this. @vlad20012 what do you think?
```
abstract class ForPostfixTemplate(provider: RsPostfixTemplateProvider, pred: (RsExpr) -> Boolean) :
    StringBasedPostfixTemplate("iter", "for x in expr", RsTopMostInScopeSelector(pred), provider) {

    override fun getElementToRemove(expr: PsiElement): PsiElement = expr
}

private val RsExpr.implementsIter: Boolean
    get() {
        val items = StdKnownItems.relativeTo(this)
        val implLookup = ImplLookup(project, items)
        val iterTrait = items.findIteratorTrait() ?: return false
        return implLookup.canSelect(TraitRef(type, iterTrait.withDefaultSubst()))
    }

private val RsExpr.implementsIntoIter: Boolean
    get() {
        val items = StdKnownItems.relativeTo(this)
        val implLookup = ImplLookup(project, items)
        val intoIterTrait = items.findIntoIteratorTrait() ?: return false
        return implLookup.canSelect(TraitRef(type, intoIterTrait.withDefaultSubst()))
    }

class IterPostfixTemplate(provider: RsPostfixTemplateProvider)
    : ForPostfixTemplate(provider, RsExpr::implementsIter) {

    override fun getTemplateString(element: PsiElement): String =
        "for x in ${element.text} {\n\$END$\n}"
}

class IntoIterPostfixTemplate(provider: RsPostfixTemplateProvider)
    : ForPostfixTemplate(provider, RsExpr::implementsIntoIter) {

    override fun getTemplateString(element: PsiElement): String =
        "for x in ${element.text}.into_iter() {\n\$END$\n}"
}
```

Reply:
It's not good idea because it leads to such situation
<img width="506" alt="screen shot 2018-09-09 at 12 59 19" src="https://user-images.githubusercontent.com/2539310/45263345-54dd0180-b430-11e8-8ec3-29f59978c248.png">
And due to [this](https://github.com/intellij-rust/intellij-rust/pull/2807#discussion_r216153042) I think it's worth to merge them again with the same template string